src/djcytoscape/tests/test_views.py

```diff
@@ -81,11 +83,62 @@ def test_all_page_status_codes_for_teachers(self):
 
+        self.assert200('djcytoscape:generate_unseeded')
+        self.assert200('djcytoscape:generate_map', kwargs={'quest_id': 1, 'scape_id': self.map.id})
+
         # These will need their own tests:
         # self.assert200('djcytoscape:regenerate', args=[self.map.id])
         # self.assert200('djcytoscape:regenerate_all')
-        # self.assert200('djcytoscape:generate_map', kwargs={'quest_id': 1, 'scape_id': 1})
-        # self.assert200('djcytoscape:generate_unseeded')
+    
+    def test_ScapeGenerateMap__POST(self):
+        """ Assert a teacher can generate a map using ScapeGenerateMapView """
+        self.client.force_login(self.test_teacher)
+
+        # generate form datas
+        content_type = ContentType.objects.filter(CytoScape.ALLOWED_INITIAL_CONTENT_TYPES).first()
+        object_ = content_type.model_class().objects.first()
+
+        form_data = generate_form_data(model_form=GenerateQuestMapForm, name='New Name')
+        form_data.update({'initial_content_object': f'{content_type.id}-{object_.id}'}) 
+
+        # response tests
+        response = self.client.post(reverse('djcytoscape:generate_unseeded'), data=form_data)
+        self.assertEqual(response.status_code, 302)
+
+        self.assertTrue(CytoScape.objects.filter(name=form_data['name']).exists())
+        
+        map_ = CytoScape.objects.get(name=form_data['name'])
+        self.assertEqual(map_.initial_content_type, content_type)
+        self.assertEqual(map_.initial_object_id, object_.id)
+
+    def test_ScapeUpdateView__POST(self):
+        """ Assert a teacher can update a map using ScapeGenerateMapView """
+        self.client.force_login(self.test_teacher)
+
+        # generate form data
+        content_type = ContentType.objects.filter(CytoScape.ALLOWED_INITIAL_CONTENT_TYPES).first()
+        object_ = content_type.model_class().objects.first()
+        
+        form_data = generate_form_data(model_form=QuestMapForm, name='Updated Name')
+        form_data.update({'initial_content_object': f'{content_type.id}-{object_.id}'}) 
+
+        # response tests
+        response = self.client.post(reverse('djcytoscape:update', args=[self.map.pk]), data=form_data)
+        self.assertEqual(response.status_code, 302)
+
+        self.assertTrue(CytoScape.objects.filter(name=form_data['name']).exists())
+
+        map_ = CytoScape.objects.get(name=form_data['name'])
+        self.assertEqual(map_.initial_content_type, content_type)
+        self.assertEqual(map_.initial_object_id, object_.id)
+
+    def test_Form_get_model_options__correct_models(self):
+        """ Quick test to see if the hardcoded model list is equal to CytoScape.ALLOWED_INITIAL_CONTENT_TYPES """
+
+        dynamically_loaded_models = [ct.model_class() for ct in ContentType.objects.filter(CytoScape.ALLOWED_INITIAL_CONTENT_TYPES)]
+        hard_coded_models = [option[0] for option in get_model_options()]
+
+        self.assertEqual(set(dynamically_loaded_models), set(hard_coded_models))
```

Comment:
Why set? set will remove doubles, which you probably don't want (i.e. if one of your list has a double, that's a problem and we want that captured by this test!)

Probably better to just make sure they are in the same order (I assume that's why you are using set?)  The order should be defined, so just make sure they match.

alternately, you can use Django's assertQuerysetEqual(qs1, qs_or_list2, ordered=False) if one of them is a queryset

Reply:
im using a set since the order is of dynamically loaded models is dependent on its id in ContentTypes. So if someone wants to change the model order in hard_coded_models it doesnt fail the test.

Ill add an extra assert to check for duplicate models

Reply:
done